src/abstractdataframe/selection.jl

```diff
@@ -0,0 +1,424 @@
+# TODO:
+# * add transform and transfom! functions
+# * add `Col` wrapper for whole column operations
+# * update documentation
+# * add tests
+
+# normalize_selection function makes sure that whatever input format of idx is it
+# will end up in one of four canonical forms
+# 1) Int
+# 2) AbstractVector{Int}
+# 3) Pair{Int, Pair{ColRename, Symbol}}
+# 3) Pair{Int, <:Pair{<:Base.Callable, Symbol}}
+# 4) Pair{AbstractVector{Int}, <:Pair{<:Base.Callable, Symbol}}
+# in this way we can easily later decide on the codepath using type signatures
+
+"""
+    ColRename
+
+A singleton type indicating that column renaming operation was requested in `select`.
+"""
+struct ColRename end
+
+normalize_selection(idx::AbstractIndex, sel) = idx[sel]
```

Comment:
Maybe it would be nice to print an error mentioning `select` for unsupported types, instead of getting an error referring to `getindex`?

Reply:
I would leave it as is. The reason is that this way we do not have to update `normalize_selection` when we would possibly change the rules of `getindex` for `AbstractIndex` (e.g. start allowing `AbstractString` for indexing columns).

But if you feel it is really important to have this I can add it.